### ISSUE TITLE
[feature request] assure no Nan is entered

### ISSUE THREAD

**Issue by ilan1987** [pytorch/pytorch#10008]
I just faced a problem that my output of a model was always Nan when in eval mode, I debugged the model to find out that inside a batchnorm layer the running variance had Nan. This is a model I loaded from somewhere.
somewhere in my code I have this line :
batchnorm.running_var = torch.from_numpy(A_np_ndarray)
I think the code could be made more idiot proof by checking inside a setter of any of the values is Nan

**Comment by soumith** [comment]
We added Anomaly Detection recently, and it is part of 0.4.1 release. While it is an expensive mode (in terms of overhead), it is provided so that you can sanity check your training / models when needed using this mode: https://pytorch.org/docs/stable/autograd.html#anomaly-detection
